templates/master/01-master-container-runtime/_base/files/crio.yaml

```diff
@@ -56,41 +63,24 @@ contents:
     # This file can change, and CRIO will automatically pick up the changes within 5 minutes.
     stream_tls_ca = ""
 
-    # file_locking is whether file-based locking will be used instead of
-    # in-memory locking
-    file_locking = false
-
     # The "crio.runtime" table contains settings pertaining to the OCI
     # runtime used and options for how to set up and manage the OCI runtime.
     [crio.runtime]
 
-    # runtime is the OCI compatible runtime used for trusted container workloads.
-    # This is a mandatory setting as this runtime will be the default one
-    # and will also be used for untrusted container workloads if
-    # runtime_untrusted_workload is not set.
-    runtime = "/usr/bin/runc"
-
-    # runtime_untrusted_workload is the OCI compatible runtime used for untrusted
-    # container workloads. This is an optional setting, except if
-    # default_container_trust is set to "untrusted".
-    runtime_untrusted_workload = ""
-
-    # default_workload_trust is the default level of trust crio puts in container
-    # workloads. It can either be "trusted" or "untrusted", and the default
-    # is "trusted".
-    # Containers can be run through different container runtimes, depending on
-    # the trust hints we receive from kubelet:
-    # - If kubelet tags a container workload as untrusted, crio will try first to
-    # run it through the untrusted container workload runtime. If it is not set,
-    # crio will use the trusted runtime.
-    # - If kubelet does not provide any information about the container workload trust
-    # level, the selected runtime will depend on the default_container_trust setting.
-    # If it is set to "untrusted", then all containers except for the host privileged
-    # ones, will be run by the runtime_untrusted_workload runtime. Host privileged
-    # containers are by definition trusted and will always use the trusted container
-    # runtime. If default_container_trust is set to "trusted", crio will use the trusted
-    # container runtime for all containers.
-    default_workload_trust = "trusted"
+    # A list of ulimits to be set in containers by default, specified as
+    # "<ulimit name>=<soft limit>:<hard limit>", for example:
+    # "nofile=1024:2048"
+    # If nothing is set here, settings will be inherited from the CRI-O daemon
+    #default_ulimits = []
+
+    # Path to the OCI compatible runtime used for trusted container workloads. This
+    # is a mandatory setting as this runtime will be the default and will also be
+    # used for untrusted container workloads if runtime_untrusted_workload is not
+    # set.
+
+    # default_runtime is the _name_ of the OCI runtime to be used as the default.
+    # The name is matched against the runtimes map below.
+    default_runtime = "runc"
```

Comment:
Personally...I think it's a much cleaner pattern to have the defaults live in the *code*, and have the config files just be commented out, or mostly empty with just a few key config options and their defaults listed.  See e.g. `/etc/systemd/journald.conf`.

So...if we made that change to crio would this cease to be an upgrade hazard?

Reply:
yeah if we commented out these fields they'd be correctly populated

Reply:
So...let's do that?

Reply:
We can probably make it work. @umohnani8 If we find no handlers and a default set, then we can default to runc. Let's see if we can make 1.14.x work without having to touch the configuration. In future iterations, we can reduce the config by using more defaults :+1: